lib/ProducerSession.cpp

```diff
@@ -136,9 +136,9 @@ namespace daq::streaming_protocol {
     {
         m_allSignals.insert(signals.begin(), signals.end());
         SignalIds signalIds;
-        for (auto signalsIter = signals.begin(); signalsIter!=signals.end(); ++signalsIter) {
-            if(signalsIter->second->isDataSignal()) {
-                signalIds.push_back(signalsIter->first);
+        for (const auto& signalsIter : signals) {
```

Comment:
The loop variable name seems a bit confusing. I wouldn't use the suffix `Iter` because it's no longer acting as an iterator. Using `for (const auto& signal : signals)` looks more consistent.

Reply:
You are right. I pushed the change.